src/lib/smartdc/generate-resolv

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
```

Comment:
This should be handled by cloud-init. Is there a reason that won't work?

Reply:
Is `cloud-init` running on the SmartOS host to provision the machine? I added a dummy `/etc/void-release` so SmartOS would provision the zone. `void` doesn't have any special scripts setup, so I figured it was the job of the client tools to do it. Let me know how I should proceed. The reason I went with just adding a dummy `void-release` was so I could use my existing SmartOS install without having to manually recompile it to add support for `gentoo-release`. That will have to be done at some point though.

Reply:
With our existing images, cloud-init runs inside the lx instance and communicates via a metadata socket to retrieve information from the global zone.

Reply:
I can't seem to find cloud-init on my SmartOS host:

    find / -iname  cloud-init

doesn't return anything.

I copied this code from the [void](https://github.com/joyent/sdc-vmtools-lx-brand/blob/master/src/lib/smartdc/void#L8) script. My plan was to make the void script call this script, but I didn't get around to it yet. I can do that now if you want. Or I'm open to other suggestions on how better to handle this.

Reply:
It needs to be installed in the guest.

Reply:
I don't think cloud-init is at play with any of our lx-images iirc.
Ubuntu and redhat flavored distros set resolvers via their brand scripts:
https://github.com/joyent/illumos-joyent/blob/efb43b6d1a736be0ece8827ac979bc1f00ecbdba/usr/src/lib/brand/lx/zone/lx_boot_zone_ubuntu.ksh#L32

Void does it via the smartdc tools as you found.  I think using the void method here is fine.  It makes it easier to generate a new image with updated tools that don't require a platform change.  Others may feel more strongly about this though.

Reply:
I modified void to call this script. I also made it a bit more robust by using a temp file.